plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,290 @@
+package logstash
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	jsonparser "github.com/influxdata/telegraf/plugins/parsers/json"
+)
+
+const sampleConfig = `
+  ## This plugin reads metrics exposed by Logstash Monitoring API.
+  # https://www.elastic.co/guide/en/logstash/current/monitoring.html
+  #
+  # logstashURL = "http://localhost:9600"
+  logstashURL = "http://localhost:9600"
+`
+const jvmStats = "/_node/stats/jvm"
+const processStats = "/_node/stats/process"
+const pipelineStats = "/_node/stats/pipeline"
+
+// Logstash Plugin Struct
+type Logstash struct {
+	LogstashURL string
+	client      *http.Client
+}
+
+// CommonData provides struct shared between all API calls
+type CommonData struct {
+	Host        string `json:"host"`
+	Version     string `json:"version"`
+	HTTPAddress string `json:"http_address"`
```

Comment:
If a field is not used, don't specify it in the structs, this way there will be less potential for an unimportant change to break the plugin.